lib/spack/llnl/util/cpu/microarchitectures.json

```diff
@@ -64,7 +64,7 @@
         "clang": {
           "versions": ":",
           "family": "x86-64",
-          "flags": "-march={family} -mcpu=generic"
+          "flags": "-march={family}"
```

Comment:
I think here, instead of deleting a valid option for linux, we should complement it with an option for Apple's Clang. Since the suffix distinguishes among different compilers I suggest we change:
```json
"clang": {
    "versions": ":",
    "family": "x86-64",
    "flags": "-march={family} -mcpu=generic"
}
```
to
```json
"clang": [
    {
        "versions": "0.0.0-apple:",
        "family": "x86-64",
        "flags": "-march={family}"
    },
    {
        "versions": ":",
        "family": "x86-64",
        "flags": "-march={family} -mcpu=generic"
    }
]
```
There are a couple of changes that are needed in the unit tests. If that's fine with you I can push the modifications to this PR (I have them as I checked beforehand if what I wanted to propose worked or not :wink: )

Reply:
Sounds good to me, thanks for checking on that!